src/org/rust/lang/annotator/RustAnnotator.kt

```diff
@@ -29,6 +28,26 @@ public class RustAnnotator : Annotator {
             is RustTypeParam -> {
                 addTextAttributes(element, holder, RustColors.TYPE_PARAMETER)
             }
+            is RustPatIdent -> {
+                if (element.bindingMode?.mut != null) {
+                    addTextAttributes(element.identifier, holder, RustColors.MUT_BINDING)
+                }
+            }
+            is RustPathPart -> {
+                val resolved = RustResolveEngine().resolve(element).resolved
```

Comment:
Yep, unfortunately Kotlin isn't flexible enough to restrict usage of `RustResolveEngine` to just `RustReference` instances.